pkg/arvo/app/metadata-hook.hoon

```diff
@@ -0,0 +1,200 @@
+::  metadata-hook: allow syncing foreign metadata
+::
+/-  *metadata-store, *metadata-hook
+/+  default-agent
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  state-zero
+  ==
+::
++$  state-zero
+  $:  %0
+      synced=(map group-path ship)
+  ==
+--
+=|  state-zero
+=*  state  -
+^-  agent:gall
+=<
+  |_  =bowl:gall
+  +*  this        .
+      hc          ~(. +> bowl)
+      def         ~(. (default-agent this %|) bowl)
+  ::
+  ++  on-init
+    [[%pass /updates %agent [our.bowl %metadata-store] %watch /updates]~ this]
+  ::
+  ++  on-save   !>(state)
+  ++  on-load   |=(=vase `this(state !<(state-zero vase)))
+  ++  on-leave  on-leave:def
+  ++  on-peek   on-peek:def
+  ++  on-arvo   on-arvo:def
+  ++  on-fail   on-fail:def
+  ++  on-poke
+    |=  [=mark =vase]
+    ^-  (quip card _this)
+    ?+  mark  (on-poke:def mark vase)
+        %metadata-hook-action
+      =^  cards  state
+        (poke-hook-action:hc !<(metadata-hook-action vase))
+      [cards this]
+    == 
+  ::
+  ++  on-watch
+    |=  =path
+    ^-  (quip card _this)
+    ?+  path        (on-watch:def path)
+        [%group *]  [(watch-group:hc t.path) this]
+    ==
+  ::
+  ++  on-agent
+    |=  [=wire =sign:agent:gall]
+    ^-  (quip card _this)
+    ?+  -.sign  (on-agent:def wire sign)
+        %kick       =^(cards state (kick:hc wire) [cards this])
+        %watch-ack  =^(cards state (watch-ack:hc wire p.sign) [cards this])
+        %fact
+      ?+  p.cage.sign  (on-agent:def wire sign)
+          %metadata-update
+        =^  cards  state
+          (fact-metadata-update:hc wire !<(metadata-update q.cage.sign))
+        [cards this]
+      ==
+    ==
+  --
+::
+|_  =bowl:gall
+++  poke-hook-action
+  |=  act=metadata-hook-action
+  ^-  (quip card _state)
+  |^
+  ?-  -.act
+      %add-owned
+    ?>  (team:title our.bowl src.bowl)
+    :-  ~
+    ?:  (~(has by synced) path.act)  state
+    state(synced (~(put by synced) path.act our.bowl))
+  ::
+      %add-synced
+    ?>  (team:title our.bowl src.bowl)
+    =/  =path  [%group path.act]
+    ?:  (~(has by synced) path.act)  [~ state]
+    :_  state(synced (~(put by synced) path.act ship.act))
+    [%pass path %agent [ship.act %metadata-hook] %watch path]~
+  ::
+      %remove
+    =/  ship  (~(get by synced) path.act)
+    ?~  ship  [~ state]
+    ?:  &(?!(=(u.ship our.bowl)) ?!((team:title our.bowl src.bowl)))
+      [~ state]
+    :_  state(synced (~(del by synced) path.act))
+    %-  zing
+    :~  (unsubscribe [%group path.act] u.ship)
+        ?.  &(=(u.ship our.bowl) (team:title our.bowl src.bowl))  ~
+        [%give %kick ~[[%group path.act]] ~]~
```

Comment:
I don't understand why we do the `team` check here, or any of these, really. If we remove it, shouldn't we always kill all subscriptions to it?

Reply:
No. We only need to kill subscriptions if we are the host of the resource.

Reply:
The `+watch-group` seems to allow incoming subscriptions on arbitrary resources though, as long as they exist in `synced`. Perhaps we want a `?>  =(our.bowl (~(got by synced) path))` there?

Reply:
The team check can be removed, but I'll add the assert.